tests/unittests/algo/test_configspace.py

```diff
@@ -0,0 +1,69 @@
+import pytest
```

Comment:
Also, it could probably be useful to also test the transformed dimensions `TransformedDimension`, `ReshapedDimension`, etc. that are passed to the constructors of the algorithms.

If you choose to support these kind of dimensions, you can use this kind of pattern to get the delegate function based on the type of the wrapped dimension, instead of the type of the wrapper:

```python
@to_configspace.register(TransformedDimension)
def _transformed_to_configspace(dim: TransformedDimension) -> HyperParameter:
    unwrapped_dim = dim.original_dimension
    # Get the handler for the unwrapped dim:
    handler_fn = to_configspace.dispatch(type(unwrapped_dim))    
    converted_unwrapped_dim = handler_fn(unwrapped_dim)
    # do something with it? Like re-add the transformation somehow?
    return converted_unwrapped_dim  # just drop the transformation for now.
```

Reply:
Looks like most of the dispatch would be done this way since algorithms receives transformed space. The use of strings to map the conversion methods may be more convenient than dispatch because of this.

Reply:
That's incorrect: https://github.com/Epistimio/orion/pull/832/files#diff-eec91bf300bfcaf8b606f6046f67669737948168c0e7a6b00133c91965f5ddd1R104

The name of the class is used for dispatching. The issue is the same.

Reply:
Indeed: https://github.com/Epistimio/orion/pull/832#discussion_r849864754